garak/detectors/base.py

```diff
@@ -72,7 +72,10 @@ def detect(self, attempt: garak.attempt.Attempt) -> Iterable[float]:
 class HFDetector(Detector, HFCompatible):
     """Detector using a Hugging Face model"""
 
-    DEFAULT_PARAMS = Detector.DEFAULT_PARAMS | {"hf_args": {"device": "cpu"}}
+    DEFAULT_PARAMS = Detector.DEFAULT_PARAMS | {
+        "hf_args": {"device": "cpu"},
+        "tokenizer_kwargs": {"padding": True, "truncation": True, "max_length": 512},
```

Comment:
I suspect `max_length` might want to come from model `ctx_len` - but this is out of scope for this PR

Reply:
I'd actually exclude `max_length` since it's very specifically model/config dependent and this might override defaults. If the model's `max_length` is, for some reason <512, this default could throw errors.

Reply:
On the other hand, running without a length cap also risks breaking models (this happens regularly on e.g. some NVCF instances). I think it's OK to offer a default control over this with a mild value that balances both being too high and too low. 512 *seems* reasonable (bearing in mind that detectors tend to be smaller models - this is a side effect of garak needing to run them at scale, but sometimes having primary compute allocated to the target model, reducing detector resources).